instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java

```diff
@@ -58,8 +58,9 @@ static void injectParent(Context context, final Map<String, Object> exchangeHead
     }
 
     @Override
-    public String get(Map<String, Object> map, String s) {
-      return (map.containsKey(s) ? map.get(s).toString() : null);
+    public String get(Map<String, Object> map, String key) {
+      Object value = map.get(key);
+      return (value == null ? null : value.toString());
```

Comment:
I agree this new style is cleaner but my eyes aren't being able to see the NPE fix - is the map bring mutated between containsKey and get previously?

Reply:
I understand that map has an explicit value `null` assigned to the given key.

Reply:
Nikita is right, previously we only checked if there is a key/value mapping and dereferenced value if so.

Reply:
Got it, thanks!